File: daily/2779. 数组的最大美丽值.cpp
class Solution {
public:
    int maximumBeauty(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int res = 0;
        for (int i = 0, j = 0; i < n; i++) {
            while (nums[i] - nums[j] > 2 * k) {
                j ++;
            }
            res = max(res, i-j+1);
        }
        return res;
    }
};

File: algotithms/python/2779. 数组的最大美丽值.py
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()
        ans = left = 0
        for right, x in enumerate(nums):
            while x - nums[left] > 2*k:
                left += 1
            ans = max(ans, right-left+1)
        return ans
        